### PR TITLE
Forms fixes

### PR DESCRIPTION
**Pull request by tpluscode**
Maybe I went a little overboard here but changing some of the form view components into custom elements make them that much more easier to manage and data-bind

### CODE REVIEW

File: ui/src/components/JobForm.vue [hunk 1]
```diff
@@ -3,14 +3,13 @@
...
-      :error="error"
-      :is-submitting="isSubmitting"
-      @submit="onSubmit"
-      :show-cancel="false"
+    <cc-hydra-operation-form
+      :operation.prop="operation"
+      :resource.prop="resource"
+      :shape.prop="shape"
+      :error.prop="error"
+      :submitting.prop="isSubmitting"
```
Comment (cristianvasquez) [inline ui/src/components/JobForm.vue:11]: The name is-submitting is more evident than `:resource.prop`

Why the change?

Comment (tpluscode) [reply ui/src/components/JobForm.vue:11]: Because vue insisted on keeping the kebab syntax as `is-submitted` and not `isSubmitted` as the property is called. the binding did not work that way

Comment (cristianvasquez) [reply ui/src/components/JobForm.vue:11]: state returning whatever the form needs to communicate.

Comment (tpluscode) [reply ui/src/components/JobForm.vue:11]: Ok, I see the comment update now and maybe I did not understand the question...

File: ui/src/customElements/FormSubmitCancel.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { createCustomElement } from '@/forms/custom-element'
+import FormSubmitCancel from '@/components/FormSubmitCancel.vue'
+
+createCustomElement('cc-form-submit-cancel')(FormSubmitCancel)
```
Comment (cristianvasquez) [inline ui/src/customElements/FormSubmitCancel.ts:4]: Just to tell you,  one can declare them globally, like https://github.com/zazuko/cube-viewer/blob/489063e6d89f4c7ab2c02345184916ad73577d1a/vue.config.js#L32

But I prefer being explicit on it, like you did :)

Comment (tpluscode) [reply ui/src/customElements/FormSubmitCancel.ts:4]: That's not the same. The line you linked is so that due does not treat custom elements as vue components.

Here it's registering the element with the browser itself

File: ui/src/components/JobForm.vue (hunk 1 above)
Comment (cristianvasquez) [reply ui/src/components/JobForm.vue:11]: I took that idea from: https://vueuse.org/core/useAsyncState/

**Review by cristianvasquez — APPROVED** [review]

File: ui/src/customElements/FormSubmitCancel.ts (hunk 2 above)
Comment (cristianvasquez) [reply ui/src/customElements/FormSubmitCancel.ts:4]: Ok, I need to read more about that. ;)

Comment (tpluscode) [reply ui/src/customElements/FormSubmitCancel.ts:4]: It wraps the native [`customElements.define()`](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define)

**Review by l00mi — COMMENTED** [review]
👍
